pkg/apis/v1alpha1/provider.go

```diff
@@ -24,10 +24,10 @@ import (
 type AWS struct {
 	// TypeMeta includes version and kind of the extensions, inferred if not provided.
 	// +optional
-	metav1.TypeMeta `json:",inline"`
+	metav1.TypeMeta `json:",inline" hash:"ignore"`
 	// AMIFamily is the AMI family that instances use.
 	// +optional
-	AMIFamily *string `json:"amiFamily,omitempty"`
+	AMIFamily *string `json:"amiFamily,omitempty" hash:"ignore"`
```

Comment:
What happens if `AMIFamily` changes but AMISelector doesn't change?

Reply:
i.e. Changing from `Bottlerocket` AMIFamily to `Custom` or vise-versa?

Reply:
That will be picked up by the AMI dynamic drift. We the AMIFaimly will alter what the resolved list of AMI IDs will be, so It will drift.

Reply:
Not necessarily. Imagine someone uses an AMISelector. Now the AMIFamily does not affect the set of resolved AMIs. Now it only affects how we treat the userData and some details of node launch like allocatable details.